### ISSUE TITLE
`TypeError: cannot pickle 'BufferedReader' instances` when creating thread

### ISSUE THREAD

**Issue by chrostino** [nnsee/payload-dumper#4]
I get this error when using the module:
```
C:\Users\Brudda\Downloads>payload_dumper payload.bin



















Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Python312\Scripts\payload_dumper.exe\__main__.py", line 7, in <module>
  File "C:\Python312\Lib\site-packages\payload_dumper\dumper.py", line 285, in main
    dumper.run()
  File "C:\Python312\Lib\site-packages\payload_dumper\dumper.py", line 91, in run
    self.multiprocess_partitions(partitions_with_ops)
  File "C:\Python312\Lib\site-packages\payload_dumper\dumper.py", line 115, in multiprocess_partitions
    process.start()
  File "C:\Python312\Lib\multiprocessing\process.py", line 121, in start
    self._popen = self._Popen(self)
                  ^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\context.py", line 337, in _Popen
    return Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\popen_spawn_win32.py", line 94, in __init__
    reduction.dump(process_obj, to_child)
  File "C:\Python312\Lib\multiprocessing\reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle 'BufferedReader' instances

Partitions   0%|                                                                          |  0/10 [00:00<?, 0.00 part/s]
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\multiprocessing\spawn.py", line 132, in _main
    self = reduction.pickle.load(from_parent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
EOFError: Ran out of input
```

**Comment by chrostino** [comment]
still get the same issue; though I don't know if the change has taken effect, I tried upgrading the module with pip but it told me it didn't need to update, I don't really know how it works

**Comment by leo-liar** [comment]
Had the same error but your commit fixed it.

<details>
  <summary>How to install:</summary>

Went to look at the specific [`commit 710eb19`](710eb19), then [`Browse files`](https://github.com/nnsee/payload-dumper/tree/710eb1995c2f656b846a651a31976dd806fee9c5) to get the whole repo in that state and then simply downloaded the [`repo zip`](https://codeload.github.com/nnsee/payload-dumper/zip/710eb1995c2f656b846a651a31976dd806fee9c5) and just installed that via `pip`.

</details>

**Comment by nnsee** [comment]
Thanks for confirming, leo! I've published an updated version (0.3.0) with the fix applied to Pypi.
